src/dxgi/dxgi_output.cpp

```diff
@@ -46,8 +46,77 @@ namespace dxvk {
     const DXGI_MODE_DESC *pModeToMatch,
           DXGI_MODE_DESC *pClosestMatch,
           IUnknown       *pConcernedDevice) {
-    Logger::err("DxgiOutput::FindClosestMatchingMode: Not implemented");
-    return E_NOTIMPL;
+    if (pModeToMatch == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: pModeToMatch is nullptr");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+        
+    if (pClosestMatch == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: pClosestMatch is nullptr");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+
+    if (pModeToMatch->Format == DXGI_FORMAT_UNKNOWN && pConcernedDevice == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: no pointer to device was provided for DXGI_FORMAT_UNKNOWN format");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+
+    if (pModeToMatch->Format == DXGI_FORMAT_UNKNOWN) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: matching formats to device currently is not supported");
+      return DXGI_ERROR_UNSUPPORTED;
+    }
+ 
+    DXGI_MODE_DESC modeToMatch = *pModeToMatch;
+    UINT modesCount = 0;
+    GetDisplayModeList(pModeToMatch->Format, 0, &modesCount, nullptr);
+
+    if (modesCount == 0) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: no device formats were found");
+      return DXGI_ERROR_NOT_FOUND;
+    }
+
+    DXGI_MODE_DESC * modes = new DXGI_MODE_DESC[modesCount];
```

Comment:
I'd like to see an `std::vector<DXGI_MODE_DESC>` here instead of a raw array.

Reply:
OK, 'll fix.